src/chat/chat_panel.rs

```diff
@@ -442,23 +445,23 @@ live_design! {
     }
 }
 
-#[derive(PartialEq)]
-enum ChatPanelState {
-    Unload {
-        downloaded_model_empty: bool,
-    },
-    Idle,
-    Streaming {
-        auto_scroll_pending: bool,
-        auto_scroll_cancellable: bool,
+#[derive(Clone, Copy, Debug)]
+enum State {
+    /// `Unknown` is simply the default state, meaning the state has not been loaded yet,
+    /// and therefore indicates a development error if it is encountered.
+    Unknown,
+    NoModelsAvailable,
+    NoModelSelected,
+    ModelSelectedWithEmptyChat,
+    ModelSelectedWithChat {
+        sticked_to_bottom: bool,
+        is_streaming: bool,
     },
 }
 
-impl Default for ChatPanelState {
-    fn default() -> ChatPanelState {
-        ChatPanelState::Unload {
-            downloaded_model_empty: true,
-        }
+impl Default for State {
+    fn default() -> Self {
+        Self::Unknown
```

Comment:
we can just derive Default and do #[default] over Unkown since the default is now a unit variant